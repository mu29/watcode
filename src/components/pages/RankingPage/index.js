import React from 'react'
import styled from 'styled-components'
import { Header, Tab, TabPage, GenericTemplate } from 'components'
import { ArtworkList } from 'containers'

const StyledTab = styled(Tab)`
  width: 48rem;
  margin: 4rem 0;

  @media(max-width: 1024px) {
    width: 90%;
  }
`

const RankingPage = () => (
  <GenericTemplate header={ <Header location="/ranking" /> }>
    <StyledTab>
      <TabPage title="오늘">
        <ArtworkList />
      </TabPage>
      <TabPage title="주간">
        <ArtworkList />
      </TabPage>
      <TabPage title="월간">
        <ArtworkList />
      </TabPage>
    </StyledTab>
  </GenericTemplate>
)

export default RankingPage
