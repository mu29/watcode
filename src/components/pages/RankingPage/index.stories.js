import React from 'react'
import { storiesOf } from '@storybook/react'
import RankingPage from '.'

storiesOf('Pages/RankingPage', module)
  .add('기본', () => (
    <RankingPage />
  ))
