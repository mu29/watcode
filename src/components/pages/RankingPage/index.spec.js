import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import RankingPage from '.'

const wrap = (props = {}) => shallow(<RankingPage { ...props } />)

describe('<RankingPage />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })
})
