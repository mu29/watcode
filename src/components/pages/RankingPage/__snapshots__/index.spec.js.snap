// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RankingPage /> 정상적으로 렌더링된다 1`] = `
<GenericTemplate
  header={
    <Header
      location="/ranking"
    />
  }
>
  <StyledTab>
    <TabPage
      title="오늘"
    >
      <Connect(ArtworkListContainer) />
    </TabPage>
    <TabPage
      title="주간"
    >
      <Connect(ArtworkListContainer) />
    </TabPage>
    <TabPage
      title="월간"
    >
      <Connect(ArtworkListContainer) />
    </TabPage>
  </StyledTab>
</GenericTemplate>
`;
